contrib/mesos/pkg/executor/registry.go

```diff
@@ -0,0 +1,305 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package executor
+
+import (
+	"encoding/json"
+	"errors"
+	"sync"
+
+	"k8s.io/kubernetes/contrib/mesos/pkg/executor/messages"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/meta"
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+
+	log "github.com/golang/glog"
+)
+
+type (
+	RegistrationEvent int
+
+	RegisteredPod struct {
+		*api.Pod
+		taskID string
+		event  RegistrationEvent
+	}
+
+	Registry interface {
+		Update(pod *api.Pod) (*RegisteredPod, error)
+		Remove(taskID string) error
+
+		bind(taskID string, pod *api.Pod) error
```

Comment:
the capital things are that way because they're invoked from the
executor/service/podsource module. otherwise i would have just made
everything private

On Fri, Jan 15, 2016 at 10:58 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/executor/registry.go
> <https://github.com/kubernetes/kubernetes/pull/19632#discussion_r49869073>
> :
>
> > +)
> > +
> > +type (
> > +	RegistrationEvent int
> > +
> > +	RegisteredPod struct {
> > +		*api.Pod
> > +		taskID string
> > +		event  RegistrationEvent
> > +	}
> > +
> > +	Registry interface {
> > +		Update(pod *api.Pod) (*RegisteredPod, error)
> > +		Remove(taskID string) error
> > +
> > +		bind(taskID string, pod *api.Pod) error
>
> Why some capital, some lowercase?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19632/files#r49869073>.
>